lib/screens/surveydetails/widget/answer_nps_widget.dart

```diff
@@ -0,0 +1,124 @@
+import 'dart:math';
+
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:lydiaryanfluttersurvey/resources/dimensions.dart';
+
+const int _npsMaxSize = 10;
+
+class AnswerNpsWidget extends StatefulWidget {
+  final int count;
+  final Function(int) onRated;
+
+  const AnswerNpsWidget({
+    Key? key,
+    required this.count,
+    required this.onRated,
+  }) : super(key: key);
+
+  @override
+  State<AnswerNpsWidget> createState() => _AnswerNpsWidgetState();
+}
+
+class _AnswerNpsWidgetState extends State<AnswerNpsWidget> {
+  int? selectedIndex;
+
+  void _setNpsValue(int index) {
+    setState(() => selectedIndex = index);
+    widget.onRated(index);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      mainAxisAlignment: MainAxisAlignment.center,
+      crossAxisAlignment: CrossAxisAlignment.center,
+      children: [
+        _buildNpsPicker(),
+        const SizedBox(
+          height: 15,
+        ),
+        _buildNpsLabels(),
+      ],
+    );
+  }
+
+  Widget _buildNpsPicker() {
+    return Container(
+      decoration: BoxDecoration(
+        border: Border.all(
+          color: Colors.white,
+          width: Dimensions.divider,
+        ),
+        borderRadius:
+            BorderRadius.circular(Dimensions.radiusRoundedRectangleButton),
+      ),
+      padding: const EdgeInsets.only(
+        left: Dimensions.paddingSmall,
+        right: Dimensions.paddingSmall,
+      ),
+      child: IntrinsicHeight(
+        child: Row(
+          mainAxisAlignment: MainAxisAlignment.spaceBetween,
+          children: _npsWidgets,
+        ),
+      ),
+    );
+  }
+
+  List<Widget> get _npsWidgets {
+    List<Widget> widgets = [];
+    for (int index = 0; index < min(widget.count, _npsMaxSize); index++) {
+      if (index != 0) {
+        widgets.add(
+          const VerticalDivider(
+            thickness: Dimensions.divider,
+            color: Colors.white,
+          ),
+        );
+      }
+      final isSelected = index <= (selectedIndex ?? -1);
+      widgets.add(
+        GestureDetector(
+          onTapDown: (details) => _setNpsValue(index),
+          child: FittedBox(
+            fit: BoxFit.contain,
+            child: Text(
+              (index + 1).toString(),
+              style: Theme.of(context).textTheme.bodyLarge?.copyWith(
+                    fontWeight:
+                        isSelected ? FontWeight.bold : FontWeight.normal,
+                    color: Colors.white.withAlpha(isSelected ? 255 : 128),
```

Comment:
```suggestion
                    color: isSelected ? Colors.white : Colors.white54,
```

Since the design used alpha 0.5, I think we can use `white54`. What do you think?

Reply:
Updated all in [28b9bb5](https://github.com/nimblehq/flutter-ic-lydia-ryan/pull/93/commits/28b9bb579d0fc94d49f159bad5c0146a599bfd2f) 🙏🏻